src/OrleansConsulUtils/ConsulBasedGatewayListProvider.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Consul;
+using Orleans.Messaging;
+using Orleans.Runtime.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Orleans.Runtime.Host
+{
+    public class ConsulBasedGatewayListProvider : IGatewayListProvider
```

Comment:
@jdom which namespace should those GatewaylistProvider live in? and the same question to membershiptables

Reply:
I think it should be `Orleans.Runtime.Membership`, but don't know for sure. The one defining this and tracking it was @attilah